src/actions/PrecursorAction.C

```diff
@@ -269,5 +311,60 @@ PrecursorAction::act()
         _problem->addInitialCondition("RandomIC", ic_name, params);
       }
     }
+
+    if (_current_task == "add_postprocessor" && getParam<bool>("loop_precs"))
+    {
+        // looping precursors requires connecting outlet of core problem
+        // to the inlet of the loop subproblem. In addition, the outlet of the
+        // loop must be connected to the core problem.
+        {
+            std::string postproc_name = "Outlet_SideAverageValue_"+var_name+"_"+_object_suffix;
+            InputParameters params = _factory.getValidParams("SideAverageValue");
+            std::vector<VariableName> varvec(1);
+            varvec[0] = var_name;
+            params.set<std::vector<VariableName>>("variable") = varvec;
+            params.set<std::vector<BoundaryName>>("boundary") =
+                getParam<std::vector<BoundaryName>>("outlet_boundaries");
+            
+            _problem->addPostprocessor("SideAverageValue", postproc_name, params);
+        }
+        {
+            std::string postproc_name = "Inlet_SideAverageValue_"+var_name+"_"+_object_suffix;
+            InputParameters params = _factory.getValidParams("Receiver");
+            params.set<MultiMooseEnum>("execute_on")= "timestep_begin";
+            
+            _problem->addPostprocessor("Receiver", postproc_name, params);
+        }
+        
+    }
+
```

Comment:
this act() function is very long. Probably best to separate into multiple smaller functions.

Reply:
Rudy Stammler's book claims that <500 lines is acceptable. :man_shrugging:

Reply:
I respectfully disagree with Rudy Stammler. Robert C. Martin's Book, clean code, suggests something more like paragraph length. So does my book, and this paper http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1001745.  There are extremely good reasons for this (testability, any function should do only one thing, DRY, and human cognition.)

Reply:
You know, I would tend to lean towards the position of someone who was a programming rather than engineering expert. This is definitely something to do in the future.

Reply:
At this point, I would tend to agree with @katyhuff. I can be pretty lazy about splitting functions but it probably is for the best...maybe something like:
```c++
if (_current_task == "add_bc")
  bcAct();
```